ArduPlane/mode_rtl.cpp

```diff
@@ -25,9 +25,27 @@ void ModeRTL::update()
     plane.calc_throttle();
 
     bool alt_threshold_reached = false;
+
+    int32_t curr_terr_alt_cm, old_alt;
+    bool alt_success;
+
     if (plane.g2.flight_options & FlightOptions::CLIMB_BEFORE_TURN) {
         // Climb to ALT_HOLD_RTL before turning. This overrides RTL_CLIMB_MIN.
-        alt_threshold_reached = plane.current_loc.alt > plane.next_WP_loc.alt;
+        if (plane.g.terrain_follow == 1) {
```

Comment:
```suggestion
        if (terrain_enabled_in_current_mode()) {
```

Reply:
Are you sure about this? IIRC I tried it but it returned 1 if terrain was enabled but not followed, and I got wrong results.